### DISCUSSION TITLE
Block Cart Editing and Content and Design Issue

### DISCUSSION THREAD

**Discussion by rohankleem** [woocommerce/woocommerce#47921] [Checkout Flow]
There's the option to provide feedback from within WooCommerce admin, which links to here. I wanted to change some content in the Block cart, but couldn't find how to, appears as though no UI to do so currently which is understandable, but also seems difficult from a development point of view. The block checkout system seems to be providing better user experience in some small ways, mostly through more ajax and client side scripts as far as I can see, but the current editability is very confusing from a development perspective and I haven't been able to do it at all. It all looks a little undercooked at the moment. When trying to change simple wording in the Cart, for example ( a core page in the user experience), I've tried in multiple files to look for and replace wording but this doesn't seem to have any effect on the content, so I've just given up. There are multiple JS files with duplicated and hard coded content across the plugin, as well as .pot files, mixed in with old templates etc. Documentation doesn't yet seem to have anything useful on how to solve this. I understand this is in development by WooCommerce with ways to customise, but it looks fraught with looming problems from my view. I'll just have to wait and see how this unfolds I guess. 

What adds pain to this is that the design choices, content and coding seems really idiosyncratic - a bit peculiar and unique - based on the personality of the person designing it almost. Particularly around high traffic areas like the cart? For example, there's a stylies smiley-face-sad icon who is crying in the middle of the cart when it's empty that is likely to be out of step with 95% of websites using it. There's things like exclamation marks after content (which completely changes the voice/attitude of the content). And clunky content like "New in store" is misleading, and once again strange choice of content for what is likely meant to be the most recently added products (which could have been added a year ago). And this is also reflected in the code, with things like lots of hardcoding, strange hard coded hex colours. 

For a new platform perhaps a "bottom-up" less opinionated design approach would be a better way to start. For example, how about just normal paragraph text "Your cart is currently empty". If an icon is needed, how about something more neutral - like cart icon. And and perhaps just a link, back to shop. That way it's likely to fit in with more stores. Keep it really simple.

![image](https://github.com/woocommerce/woocommerce/assets/6668421/a1b23cdc-4e07-4bca-b736-8d35a6ce369e)

In case you were able to help, then I guess the thing that I am looking to understand and change, is which files and elements I can change to undo all these changes and just have it as a neutral page, I'll take care of the rest in my store shell. Thanks very much.

**Comment by nielslange** [comment]
Thanks for reaching out, @rohankleem. It appears that it's not obvious how to edit the page, that is visible, when the cart if empty. You can edit that page directly in the page editor, by opening the document overview and then selecting the Empty Cart block:

<img width="1277" alt="Screenshot 2024-05-29 at 16 39 42" src="https://github.com/woocommerce/woocommerce/assets/3323310/b14921e4-39c6-46cf-a82e-f1ab7eaa7a53">

The icon had been added using CSS. The following two CSS snippets allow removing the icon:

```css
// This removes the icon and all it's spacing:
.wc-block-cart__empty-cart__title.with-empty-cart-icon::before { display: none; }

// This only removed the icon, but keeps the spacing:
.wc-block-cart__empty-cart__title.with-empty-cart-icon::before { visibility: hidden; }
```

That said, it would be nice to be able to edit or remove the icon directly in the editor. I've created #47937 for tracking this.

The `!` behind `Your cart is currently empty!` can be removed directly in the page editor. As for `New in store`, that section displays the Newest Products block. Both the title and the Newest Products can be replaced with different content, e.g. `Products on sale`:

<img width="1277" alt="Screenshot 2024-05-29 at 16 41 32" src="https://github.com/woocommerce/woocommerce/assets/3323310/7043a937-16bf-4fa5-9162-7089dd3c08cc">

As for the different styling of the sales badges, can you show an example where you see different styles? On my end, all `sales` badges seem to use the same styles.

Regarding the font sizes, I noticed that they depend on the number of columns. I created #47939 for tracking this issue.

You also mentioned that the button styles look inconsistent. On my end, the buttons look consistent across the various pages. Is it possible that you;re seeing a different result based on your theme?

**Reply by rohankleem** [reply]
Ok, thanks, didn't realise that's how some could be edited. Regarding the sale icon, below shows this. Normally the button below my product tiles is Yellow, but to be fair I have removed that from the tile list pages. 
![image](https://github.com/woocommerce/woocommerce/assets/6668421/2105c0aa-1b8a-4e8a-80ec-88fadf5ceb7c)

**Reply by nielslange** [reply]
Thanks for sharing a screenshot of the `sales` badges. This design seems to be applied by your current theme. The default `sales` badge should look like in the screenshot I shared above. Could you temporarily switch to a different theme, e.g. the [Twenty Twenty-Four](https://wordpress.org/themes/twentytwentyfour/) theme to double-check this?

**Reply by rohankleem** [reply]
On checking out the code that dictates these styles, the only references I can find are all files directly in the woocommerce plugin, with hardcoded values not relating to any themes, for example like "background-color:#b3af54;", see attached screenshot. I submit that is one aspect which is not good about general WooCommerce styling, but to the other point about these in the cart, the HTML used for the tiles is different to product tiles used in archive pages, which speaks to the other problem of inconsistency of elements used , so even if they were custom styled, the styling only applies to one unless two classes are setup for the two different html snippets. In this example, the cart uses completely different css classes, the .wc-block-grid class prefix to most elements, which I guess given the transition to blocks is understandable, but makes development tough.

![image](https://github.com/woocommerce/woocommerce/assets/6668421/40b296bb-8361-465d-80f2-a2f1c0d2a48b)
![image](https://github.com/woocommerce/woocommerce/assets/6668421/1752ac72-2672-497d-aa78-87feac799ae2)
![image](https://github.com/woocommerce/woocommerce/assets/6668421/42f4881e-0848-4e71-8f4c-9d72da5e1a96)

**Reply by nielslange** [reply]
Can you share your [system status report](https://woocommerce.com/document/understanding-the-woocommerce-system-status-report/) with us or tell us which theme you are using so that we can reproduce this issue?

**Comment by jocala99** [comment]
I agree that Cart block customization is baffling for developers. I simply need to change the heading text "Products" to the word "Items" and I can find no way to do it.

**Reply by nielslange** [reply]
@jocala99 Please share a screenshot of the sections you are referring to.

**Comment by jocala99** [comment]
I want the circled word to be ITEM instead of PRODUCT.
![Screenshot 2024-07-15 at 7 32 09 AM](https://github.com/user-attachments/assets/59054c1b-94fb-4744-8a79-f24df648936d)

**Comment by nielslange** [comment]
Thanks for sharing the screenshot, @jocala99. Currently, it's not possible to edit that string within the editor. I just created [[Enhancement]: Enable merchants to edit the "Product" and "Total" texts in the Cart block](https://github.com/woocommerce/woocommerce/issues/49559) to make the "Product" and the "Total" texts editable.

For now, you could use the [Say What?](https://wordpress.org/plugins/say-what/) plugin for editing the text. Here are the steps you'd have to take:

1. Install and activate the Say What? plugin.
2. Go to `/wp-admin/tools.php?page=say_what_admin&say_what_action=addedit`.
3. Provide the following information:

- Original string: **Product**
- Text domain: **woocommerce**
- Text content: _you can leave this field empty_
- Replacement string: **Item**

The form should look like this:

<img width="1023" alt="Screenshot 2024-07-15 at 16 55 03" src="https://github.com/user-attachments/assets/e42ba401-6c49-4cb8-a3d5-7c9831ac64a0">

This then leads to the following result on the Cart block:

<img width="1095" alt="Screenshot 2024-07-15 at 16 57 16" src="https://github.com/user-attachments/assets/0f5cf676-027d-46b4-ba74-4cbccb5df852">

**Reply by jocala99** [reply]
Thanks @nielslange!

**Reply by nielslange** [reply]
You're very welcome, @jocala99.
